test/app-tap/module_api.c

```diff
@@ -347,6 +347,31 @@ test_key_def_api(lua_State *L)
 	return 1;
 }
 
+static int
+check_error(lua_State *L)
+{
+	uint32_t fieldno1[] = {0};
+	uint32_t type1[] = {FIELD_TYPE_STRING};
+	uint32_t fieldno2[] = {0};
+	uint32_t type2[] = {FIELD_TYPE_UNSIGNED};
+	box_key_def_t *key_defs[] = {
+		box_key_def_new(fieldno1, type1, 1),
+		box_key_def_new(fieldno2, type2, 1)};
+	box_tuple_format_t *format = box_tuple_format_new(key_defs, 2);
+	if (format != NULL) {
+		box_tuple_format_unref(format);
+		box_key_def_delete(key_defs[0]);
+		box_key_def_delete(key_defs[1]);
+		lua_pushboolean(L, false);
+		return 1;
+	}
+	luaT_error(L);
```

Comment:
luaT_error() raises an exception and unwinds C stack, so the code below is not called.
Please simplify your test case by using box_set_error() + luaT_error() instead.